src/test/java/uk/gov/hmcts/reform/laubackend/cases/controllers/AccessRequestControllerTest.java

```diff
@@ -75,4 +80,56 @@ void shouldReturnInternalServerErrorForBadPostRequest() {
         verify(accessRequestService, times(1)).save(accessRequestLog);
         assertThat(response.getStatusCode()).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
     }
+
+    @Test
+    void shouldReturnResponseEntityForGetRequest() {
+        AccessRequestGetResponse response = AccessRequestGetResponse.builder()
+            .build();
+
+        when(accessRequestService.getAccessRequestRecords(any())).thenReturn(response);
+
+        AccessRequestGetRequest request = AccessRequestGetRequest.builder()
+            .caseRef("123")
+            .startTimestamp("2024-06-23T22:20:05.000")
+            .endTimestamp("2024-06-24T22:20:05.000")
+            .build();
+        ResponseEntity<AccessRequestGetResponse> responseEntity = accessRequestController.getAccessRequest(
+            null,
+            null,
+            request
+        );
+
+        verify(accessRequestService, times(1)).getAccessRequestRecords(request);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
```

Comment:
This test should fail as userId or requestType is not provided which is mandatory

Reply:
In my understanding, one or more must be present of those three, meaning if caseRef is there (and both timestamps) it should be fine.

Reply:
That's fine